### PR TITLE
[Feature]: Analytics add UserID

### PR DESCRIPTION
**Pull request by marcomaddy1982**
## What's new in this PR?

in order to identify an user on Analytics we added an unique id in core data for the User object

## Dependencies

JIRA: https://wearezeta.atlassian.net/browse/ZIOS-13750

### CODE REVIEW

**Comment by johnxnguyen** [comment]
Did you mean to set the target branch to `feature/analytics`?

**Review by johnxnguyen — DISMISSED** [review]

**Review by johnxnguyen — CHANGES_REQUESTED** [review]
Please provide a simple test to ensure the self user has the id set when created.

**Comment by marcomaddy1982** [comment]
@zenkins release minor

**Comment by johnxnguyen** [comment]
@marcomaddy1982 did you add the test?

File: Source/Model/User/ZMUser.m [hunk 1]
```diff
@@ -707,6 +707,7 @@ + (ZMUser *)createSessionIfNeededInContext:(NSManagedObjectContext *)moc withSel
     
     if (selfUser == nil) {
         selfUser = [ZMUser insertNewObjectInManagedObjectContext:moc];
+        selfUser.analyticsIdentifier = [[NSUUID UUID] UUIDString];
```
Comment (typfel) [inline Source/Model/User/ZMUser.m:710]: This would converted into a `transportString`

**Pull request merged**